api/urls.py

```diff
@@ -98,7 +98,16 @@ def register_url(self, subclass, viewset):
     url(r'atom/?', views.CreativeWorksAtom(), name='atom'),
     url(r'graph/?', GraphQLView.as_view(graphiql=True)),
     url(r'userinfo/?', ensure_csrf_cookie(views.ShareUserView.as_view()), name='userinfo'),
-    url(r'search/(?!.*_bulk\/?$)(?P<url_bits>.*)', csrf_exempt(views.ElasticSearchView.as_view()), name='search'),
+
+    # only match _count, _search, _suggest requests
+    url(
+        r'search/(?!.*_bulk\/?$)(?P<url_bits>(_|[a-zA-Z]+/_)((?=search)|(?=count)|(?=suggest))[a-zA-Z]{5,7}/{0,1}$)',
+        csrf_exempt(views.ElasticSearchView.as_view()),
+        name='search'
+    ),
+    # match _mappings requests
+    url(r'search/(?P<url_bits>_mappings(/.+|$|/))', csrf_exempt(views.ElasticSearchView.as_view()), name='search'),
```

Comment:
`_mappings` requests have a different format then the other whitelisted urls. I can try and put it in the same regex but I think it would be messy.